compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -128,7 +128,10 @@ fn instrument_function_attr<'ll>(cx: &CodegenCx<'ll, '_>) -> SmallVec<[&'ll Attr
 
         // The function name varies on platforms.
         // See test/CodeGen/mcount.c in clang.
-        let mcount_name = cx.sess().target.mcount.as_ref();
+        let mcount_name = match cx.sess().target.arch.as_ref() {
+            "arm" => "llvm.arm.gnu.eabi.mcount",
```

Comment:
instead of matching here, you should change the `mcount` field in the target specs in `compiler/rustc_target`

Reply:
@Nilstrieb I don't want to change the target spec because the intrinsic is LLVM-only, so it would not work on GCC or Cranelift.
Is it fine to ignore the other backends and guess that LLVM always builds the ARM target?

Reply:
I don't think it would be okay to change `mcount` to an LLVM-specific value, for reasons you've mentioned. 

However, I'm not sure how I feel about special-casing a different value in `codegen_llvm` either.

Perhaps add a `llvm_mcount_intrinsic: Option<StaticCow<str>>` field and use that when it is provided? Just add a documentation comment with it explaining that it takes precedence and is only used by LLVM codegen, etc.

Reply:
@davidtwco Thanks for your review!
I agree that the suggested code architecture is more good to see, but I'm not sure that LLVM uses mcount intrinsic except the `arm.gnu.eabi.mcount`.

However, to handle the unexpected case, I changed the commit as your suggestion.